draft-regext-rdap-extensions.md

```diff
@@ -379,6 +381,59 @@ extension registration that attempts to use one of the RDAP
 conformance values given in this section as an extension identifier
 (and so as an RDAP conformance value also) will be rejected.
 
+# Redirects {#redirects}
+
+[@!RFC7480] describes the use of redirects in RDAP. Redirects are prominent
+in the discovery of authoritative INR servers as the process outlined in
+[@?RFC9224], which uses IANA allocations, does not account for transfers of
+resources between INRs. [@!RFC7480, Section 4.3] instructs servers to ignore
+unknown query parameters. As it relates to issuing URLs for redirects, servers
+MUST NOT blindly copy query parameters from a request to a redirect URL as
+query parameters may contain sensitive information, such as security credentials,
+not relevant to the target server of the URL. Following the advice in [@!RFC7480],
+servers SHOULD only place query paramters in redirect URLs when it is known
+by the origin server (the server issuing the redirect) that the target server
+(the server referenced by the redirect) can process the query parameter and the
+contents of the query parameter are appropriate to be received by the target.
```

Comment:
If the mandatory behaviour were instead "do not copy any query parameters into the redirect URL", then things will be more consistent, since it won't depend on a server's (possibly peculiar/incorrect/changeable) understanding of which query parameters are appropriate for passing on.  It would also make it clear that clients have responsibility for amending the redirect URL if they want certain query parameters to be passed on to the server that's actually handling the query.

Reply:
That would then render things like OpenID Connect with identity servers as out of compliance. I am also not a fan of placing complexity on clients. It's already problematic with the redacted extension.

Reply:
> That would then render things like OpenID Connect with identity servers as out of compliance.

How would that be out of compliance?  All of the interactions in that document appear to involve scenarios where the client is logging in to a specific RDAP server, rather than being redirected from one RDAP server to another.

> I am also not a fan of placing complexity on clients. It's already problematic with the redacted extension.

But if servers operate inconsistently, which is likely (at least in the short term), clients may end up having to manage that complexity in some way anyway, and/or being surprised/annoyed when server B doesn't work in the same way as server A.

Reply:
OpenID Connect is about federated identity services (i.e. using your google account to authenticate to GitHub). I can't speak to that draft, but if it is not about 3rd party identity providers then it is a particularly complicated way to do authentication and authorization.

I don't think a blanket statement of don't use query parameters across authority boundaries is gonna fly. However, I think it is good to say only do it if you know the target understands the content and should be receiving it.

Reply:
> OpenID Connect is about federated identity services (i.e. using your google account to authenticate to GitHub). I can't speak to that draft, but if it is not about 3rd party identity providers then it is a particularly complicated way to do authentication and authorization.

OK, my reading of the text was such that I thought it was limited to the case of an RDAP server redirecting to another RDAP server for the purpose of responding to a standard query, and that things like the OIDC flow were out of scope.  I think it would be possible to limit this text to the case of a 'standard' redirect, and then require that query string parameters not be added to such redirects, which should address this concern (though there's still the client complexity issue).  (If the behaviour here remains per the current text, then it may still be worth clarifying there are different classes of redirects, and that the OIDC flow is an example of a redirect where the initial server knows that certain query parameters can be included in the redirect sent to the client.)

Reply:
Sorry. I think I muddied the waters. It is limited to RDAP queries. That said I think the text works as is in that query parameters can only be used in scenarios where it is known they will work.  A blanket prohibition seems to me to be too restrictive. I'd hate to freeze out some future use we don't know about. This guidance of only use them when it is known they will work coupled with the advice in the paragraphs below strikes the right balance, IMHO.

Reply:
Makes sense, thanks.